src/VS4Mac.TextGenerator/Views/GenerateTextDialog.cs

```diff
@@ -1,144 +1,174 @@
-                WidthRequest = 48
+                WidthRequest = 48
+            };
+
+            _typesBox = new HBox();
+            _typesLabel = new Label("Generator type:");
+
+            _generatorTypesComboBox = new ComboBox
+            {
+                WidthRequest = 150
             };
 
+            _generatorTypesComboBox.Items.Add("Lorem Ipsum");
+            _generatorTypesComboBox.Items.Add("Jeffsum");
+            _generatorTypesComboBox.SelectedIndex = 0;
+
             _optionsBox = new HBox();
             _optionsLabel = new Label("Generator option:");
 
-            _generatorOptionsComboBox = new ComboBox
-            {
-                WidthRequest = 150
+            _generatorOptionsComboBox = new ComboBox
+            {
+                WidthRequest = 150
             };
 
             _generatorOptionsComboBox.Items.Add("Characters");
             _generatorOptionsComboBox.Items.Add("Words");
             _generatorOptionsComboBox.Items.Add("Sentences");
-            _generatorOptionsComboBox.SelectedIndex = 1;
-        }
-
-        void BuildGui()
+            _generatorOptionsComboBox.SelectedIndex = 1;
+        }
+
+        void BuildGui()
         {
-            _optionsBox.PackStart(_optionsLabel);
+            _typesBox.PackStart(_typesLabel);
+            _typesBox.PackEnd(_generatorTypesComboBox);
+
+            _optionsBox.PackStart(_optionsLabel);
             _optionsBox.PackEnd(_generatorOptionsComboBox);
-
-            _numberBox.PackStart(_numberLabel, true);
-            _numberBox.PackEnd(_numberEntry, false);
-            _buttonBox.PackEnd(_generateButton);
 
-            _mainBox.PackStart(_optionsBox);
+            _numberBox.PackStart(_numberLabel, true);
+            _numberBox.PackEnd(_numberEntry, false);
+            _buttonBox.PackEnd(_generateButton);
+
+            _mainBox.PackStart(_typesBox);
+            _mainBox.PackStart(_optionsBox);
             _mainBox.PackStart(_numberBox);
             _mainBox.PackEnd(_buttonBox);
 
-            _numberEntry.SetFocus();
-
-            Content = _mainBox;
-            Resizable = false;
-        }
-
-        void AttachEvents()
+            _numberEntry.SetFocus();
+
+            Content = _mainBox;
+            Resizable = false;
+        }
+
+        void AttachEvents()
         {
-            _numberEntry.Changed += OnNumberEntryChanged;
+            _numberEntry.Changed += OnNumberEntryChanged;
             _generateButton.Clicked += OnGenerateClicked;
-            _generatorOptionsComboBox.SelectionChanged += OnGeneratorOptionsChanged;
-        }
-
-        void OnGeneratorOptionsChanged(object sender, EventArgs e)
-        {
-            _numberLabel.Text = $"Introduce number of {_generatorOptionsComboBox.SelectedItem.ToString().ToLower()}:";
+            _generatorOptionsComboBox.SelectionChanged += OnGeneratorOptionsChanged;
+            _generatorTypesComboBox.SelectionChanged += OnTypeOptionsChanged;

```

Comment:
Probably will include a preference pane in future versions of the addin and will include a "favourite" generator option to auto select and simplify the UI but for now, is nice.